modules/codegen/src/smithy4s/codegen/Codegen.scala

```diff
@@ -116,7 +117,7 @@ object Codegen { self =>
         namespaces
           .filterNot(_.startsWith("aws."))
           .filterNot(_.startsWith("smithy."))
-          .filterNot(reserved)
+          .filterNot(ns => reserved.exists(ns.startsWith))
```

Comment:
Changed to this so that if reserved contains `alloy` it will also filter out `alloy.proto` and future additions we may make.